src/picongpu/include/particles/creation/creation.kernel

```diff
@@ -0,0 +1,299 @@
+ *         instance of the particle creation functor
+ */
+template<class ParBoxSource, class ParBoxTarget, class ParticleCreator, class Mapping>
+__global__ void kernelCreateParticles(ParBoxSource sourceBox,
+                                      ParBoxTarget targetBox,
+                                      ParticleCreator particleCreator,
+                                      Mapping mapper)
+{
+
+    /* "particle box" : container/iterator where the particles live in
+     * and where one can get the frame in a super cell from
+     */
+    typedef typename ParBoxSource::FrameType SOURCEFRAME;
+    typedef typename ParBoxTarget::FrameType TARGETFRAME;
+    typedef typename ParBoxSource::FramePtr SourceFramePtr;
+    typedef typename ParBoxTarget::FramePtr TargetFramePtr;
+
+    /* specify field to particle interpolation scheme */
+    typedef typename PMacc::traits::Resolve<
+        typename GetFlagType<SOURCEFRAME,interpolation<> >::type
+    >::type InterpolationScheme;
+
+    /* margins around the supercell for the interpolation of the field on the cells */
+    typedef typename GetMargin<InterpolationScheme>::LowerMargin LowerMargin;
+    typedef typename GetMargin<InterpolationScheme>::UpperMargin UpperMargin;
+
+    /* relevant area of a block */
+    typedef SuperCellDescription<
+        typename MappingDesc::SuperCellSize,
+        LowerMargin,
+        UpperMargin
+        > BlockDescription_;
+
+    /* for not mixing operations::assign up with the nvidia functor assign */
+    namespace partOp = PMacc::particles::operations;
+
+    /* definitions for domain variables, like indices of blocks and threads */
+    typedef typename BlockDescription_::SuperCellSize SuperCellSize;
+    /* multi-dimensional offset vector from local domain origin on GPU in units of super cells */
+    const DataSpace<simDim> block(mapper.getSuperCellIndex(DataSpace<simDim > (blockIdx)));
+
+    /* multi-dim vector from origin of the block to a cell in units of cells */
+    const DataSpace<simDim > threadIndex(threadIdx);
+    /* conversion from a multi-dim cell coordinate to a linear coordinate of the cell in its super cell */
+    const int linearThreadIdx = DataSpaceOperations<simDim>::template map<SuperCellSize > (threadIndex);
+
+    /* multi-dim offset from the origin of the local domain on GPU
+     * to the origin of the block of the in unit of cells
+     */
+    const DataSpace<simDim> blockCell = block * SuperCellSize::toRT();
+
+    /* subtract guarding cells to only have the simulation volume */
+    const DataSpace<simDim> localCellIndex = (block * SuperCellSize::toRT() + threadIndex) - mapper.getGuardingSuperCells() * SuperCellSize::toRT();
+
+    __shared__ typename PMacc::traits::GetEmptyDefaultConstructibleType<SourceFramePtr>::type sourceFrame;
+    __shared__ typename PMacc::traits::GetEmptyDefaultConstructibleType<TargetFramePtr>::type targetFrame;
+    __shared__ lcellId_t maxParticlesInFrame;
```

Comment:
Please use a `static const variable` because this variable is always set to `PMacc::math::CT::volume<SuperCellSize>::type::value;`